### ISSUE TITLE
Move SpecVersion and SpecAssertion to general TCK utils project

### ISSUE THREAD

**Issue by arjantijms** [jakartaee/cdi-tck#592]
The CDI TCK uses `org.jboss.test.audit.annotations.SpecAssertion` and `org.jboss.test.audit.annotations.SpecVersion`. Maybe it would be a good idea to move these to the general `jakara.tck` util project?

**Comment by manovotn** [comment]
I think this is loosely tied to https://github.com/jakartaee/cdi-tck/issues/259

I.e. how we track which test file is tied to which assertion. This is still in place but slowly being more and more outdated.
Sadly, this is an item on the agenda that nobody seems to have time to work on.

> Maybe it would be a good idea to move these to the general jakara.tck util project?

Maybe we should instead think of an overall idea how to link tests to assertions across Jakarta TCKs and place that into the util project? Don't get me wrong, I don't mind the annotations being in whichever project; I just think a more complex approach than two annotations might be appropriate.

**Comment by manovotn** [comment]
> My experience with updating the TCK audit files is that the current CDI TCK model doesn't work well enough.

The original model was able to generate a whole spec text annotated with direct links to tests which was pretty neat.
That said, I still mostly used just the generated sections file along with audit files to determine exactly which tests are linked to which specification statements.

Still, the upkeep for it is definitely non-zero even as-is.

**Comment by Ladicek** [comment]
Yeah, but the original model also required (and still requires) creating a second [near] copy of the entire specification, which is quite bad (unless we have a person fully dedicated to the TCK, which we no longer do). I _think_ there's a way to lessen the overhead, but as mentioned, it needs research.
